src/Engine/GraphLayout/GraphLayout.cs

```diff
@@ -256,18 +323,29 @@ public void AssignLayers()
                     layerWidth = 0;
 
                     if (n.LeftEdges.Count == 0)
+                    {
                         isFinalLayer = true;
+                    }
                 }
 
                 AddToLayer(n, currentLayer);
                 processed++;
 
                 if (n.Width > layerWidth)
+                {
                     layerWidth = n.Width;
+                }
             }
 
-            // Put the input nodes on the leftmost layer
-            AddToLayer(Nodes.Where(x => x.LeftEdges.Count == 0).ToList(), Layers.Count);
+            // Put all input nodes and isolated nodes on the leftmost layer
+            AddToLayer(Nodes.Where(x =>
+                (x.LeftEdges.Count == 0 && !x.HasUnconnectedInPort) ||
+                (x.LeftEdges.Count == 0 && x.RightEdges.Count == 0)).ToList(), Layers.Count);
+
+            // Assign nodes with unconnected input ports right behind its next layer
+            foreach (Node n in Nodes.Where(x => x.LeftEdges.Count == 0 && x.RightEdges.Count > 0 && x.HasUnconnectedInPort))
```

Comment:
@aparajit-pratap `HasUnconnectedInPort` refers to having an input port which is not attached to any connector. The property applies true for both single function object nodes and nodes using default values.